admin-tools/lib/src/main/scala/com/gu/mediaservice/ImagesBatchProjection.scala

```diff
@@ -4,49 +4,60 @@ import java.net.URL
 
 import com.gu.mediaservice.model.Image
 
-import scala.concurrent.duration.Duration
+import scala.concurrent.duration.{DAYS, Duration, HOURS}
 import scala.concurrent.{Await, ExecutionContext, Future}
 
 class ImagesBatchProjection(apiKey: String, timeout: Duration, gridClient: GridClient, maxSize: Int) {
 
   private implicit val ThrottledExecutionContext = ExecutionContext.fromExecutor(java.util.concurrent.Executors.newFixedThreadPool(5))
 
-  def validApiKey(projectionEndpoint: String) = {
+  // TODO I don't like this - checking the API key by making a request to a non-existent endpoint is ... ewww.
```

Comment:
This is done differently in  https://github.com/guardian/grid/pull/3187/files#diff-8714f6045b0cd0fce66f948140ad54deff14c89eab51ed1597d657111207c337R28 

(Personally not sure the client wants a validate method and should instead just log the first http request that goes awry for this)

Reply:
imageloader calls are implemented differently.  I don't know why.  The  imageloaderendpoint is provided explicitly, for example.

Reply:
ah so we can't use the endpoint from there :( I see